### DISCUSSION TITLE
Too good to be true

### DISCUSSION THREAD

**Discussion by NekoTheWeird** [Discord-Oxygen/Discord-Oxygen#28] [General]
Hey, so I have stumbled upon this project a few months back. I must say, this seems really too good to be true.

It boggles my mind that something even remotely like this would be possible.
I would have thought that lot of the features of Discord Oxygen would be server checked on Discords side.

If possible may you explain it in some ways how are you able to do something like this? (I am not asking for code, just some basic theory without sharing anything major as not to be stolen)
It's hard to believe that these kind of exploits exist.

Of course I very much so hope you are able to bring this project to reality, yet sadly I cant help but doubt it.

Have a nice day.

--Neko--

**Comment by hxr404** [comment]
Hey there,
thanks for being interested in this project.

> I would have thought that lot of the features of Discord Oxygen would be server checked on Discords side.

While working on Discord Oxygen's predecessor, "the Framework", I was also kinda surprised how little Discord's backend cares about the validity of the requests. (I even submitted some exploits to Discord's bug bounty program which resulted in my Account getting banned)
But unfortunately I have to disappoint you, the most impressive features are mostly client-side only.

From the README:
> Many features listed here are client-side only!
> This means that some things like a custom discriminator or animated profile picture/banner are **only visible to other Discord Oxygen users!**

In theory, sending spoofed data to other clients (in order to display things like animated pfps) isn't that hard, the hard part is to 1. make it reasonably secure 2. make it privacy-friendly 3. ensure high-availability.
I'm experimenting with p2p-based communication (because I don't want to maintain a whole kubernetes cluster for running a new Discord...), but I'm still looking for ways to hide the users IP Adress (for obvious privacy reasons) and bundling a whole i2p or tor runtime isn't going to work.

So perhaps, under-the-hood some of the stuff mentioned on the README probably doesn't really work in a way you'd expect.

I hope this addresses most of your questions, if not, please let me know.

**Reply by KRADOWPIMP** [reply]
may u please make a new discord server because i cant trust your hack when my browser is warning me like crazy

**Comment by NekoTheWeird** [comment]
> In theory, sending spoofed data to other clients

Oh, i haven't thought of it that way. Clever.

> From the README

I missed the part about the "Only visible on other compatible clients". I apologize.

Actually that's exactly and everything what i wanted to know/was curious about.

Thank you very much in replying in such a manner. 
From this reply i will consider supporting this project. I am very aware that this is in no way an easy project, but i give you my best in succeeding.

--Neko--
